tests/sklearn/test_sklearn_models.py

```diff
@@ -1735,20 +1732,14 @@ def check_for_divergent_predictions(x, model, fhe, max_iterations=N_ALLOWED_FHE_
     fhe_diff_found = check_for_divergent_predictions(x, model, fhe="execute")
 
     # Check for differences in predictions
-    if is_linear_model:
-
-        # Linear models should give the same results whatever the p_error
-        assert fhe_diff_found
-
-        # Simulation and FHE differs with very high p_error on leveled circuit
-        # FIXME https://github.com/zama-ai/concrete-ml-internal/issues/4343
-        assert not simulation_diff_found
+    if error_param["p_error"] > 0.5:
```

Comment:
seems like a mgic number, I would prefer to modify the test input like : 
```
@pytest.mark.parametrize(
    "p_error, expected_diff",
    [(1 - 2**-40, True), (2**-40, False)],
    ids=["p_error_high", "p_error_low"],
)
``` 

and do `if expected_diff:` here

Reply:
Agreed

Reply:
Done